test-clients/src/main/java/com/hedera/services/yahcli/commands/signedstate/SignedStateHolder.java

```diff
@@ -0,0 +1,253 @@
+     * bytecodes for all the contracts in that state.
+     */
+    public static @NonNull Contracts getContracts(@NonNull Path inputFile) throws Exception {
+        try (final var signedState = new SignedStateHolder(inputFile)) {
+            final var contractIds = signedState.getAllKnownContracts();
+            final var contractContents = signedState.getAllContractContents(contractIds);
+            return new Contracts(contractContents, contractIds.size());
+        }
+    }
+
+    public @NonNull ServicesState getPlatformState() {
+        return platformState;
+    }
+
+    /** Gets all existing accounts */
+    public @NonNull AccountStorageAdapter getAccounts() {
+        final var accounts = platformState.accounts();
+        assertSignedStateComponentExists(accounts, "accounts");
+        return accounts;
+    }
+
+    /**
+     * Returns the file store from the state
+     *
+     * <p>The file state contains, among other things, all the contracts' bytecodes.
+     */
+    public @NonNull VirtualMap<VirtualBlobKey, VirtualBlobValue> getFileStore() {
+        final var fileStore = platformState.storage();
+        assertSignedStateComponentExists(fileStore, "fileStore");
+        return fileStore;
+    }
+
+    /**
+     * Returns all contracts known via Hedera accounts, by their contract id (lowered to an Integer)
+     */
+    public @NonNull Set</*@NonNull*/ Integer> getAllKnownContracts() {
+        var ids = new HashSet<Integer>();
+        getAccounts()
+                .forEach(
+                        (k, v) -> {
+                            if (null != k && null != v && v.isSmartContract())
+                                ids.add(k.intValue());
+                        });
+        return ids;
+    }
+
+    /** Returns the bytecodes for all the requested contracts */
+    public @NonNull Collection</*@NonNull*/ Contract> getAllContractContents(
+            @NonNull Collection</*@NonNull*/ Integer> contractIds) {
+
+        final var fileStore = getFileStore();
+        var codes = new ArrayList<Contract>();
+        for (var cid : contractIds) {
+            final var vbk = new VirtualBlobKey(Type.CONTRACT_BYTECODE, cid);
+            if (fileStore.containsKey(vbk)) {
+                final var blob = fileStore.get(vbk);
+                if (null != blob) {
+                    final var c = new Contract(Set.of(cid), blob.getData());
+                    codes.add(c);
+                }
+            }
+        }
+        return codes;
+    }
+
+    // (Class doesn't own any true resources, thus doesn't _need_ to be `Closeable` ... but it
+    // _does_ hold a ginormous chunk of memory.  So it is `Closeable` just to remind the user
+    // that it is expensive and should be released as soon as it is no longer needed.)
+    @Override
+    public void close() throws IOException {
+        platformState = null;
+        System.gc(); // hint to JVM
```

Comment:
I understand the reasoning here but generally calling the gc is considered bad form.  One reason is that the specific garbage collector being used is not known and thus the behavior of this call is unspecified.  Generally, rightly or wrongly most people leave the garbage collector to do it's work when it deems best.

Reply:
ok